packages/event-listener/test/index.test.ts

```diff
@@ -7,7 +7,7 @@ describe("createEventListener", () => {
       createRoot(dispose => {
         const target = new EventTarget();
         const testEvent = new Event("test");
-        createEventListener<{ test: Event }>(target, "test", ev => {
+        createEventListener<{ test: Event }, EventTarget>(target, "test", ev => {
```

Comment:
That shouldn't be necessary.

Reply:
Hi @atk :). I'm seeing this in VS Code:

![Screenshot from 2021-11-29 14-15-50](https://user-images.githubusercontent.com/297678/143951403-e97d4452-cd4a-42ce-851b-ea46d32c0f25.png)

In general we need a pass at ensuring Solid and primitives are fully aligned on TS configuration, so that we'll get the same errors in build as with VS Code (currently not the case). That'll help clear things up.

This will involve making Solid follow a conventional TypeScript project layout and unduplicating some of its type definitions.